### ISSUE TITLE
Confusing diagnostics when method fails before argument validity is checked (e.g. in `sorted`)

### ISSUE THREAD

**Issue by lucifer1004** [typst/typst#1367]
### Description

The following code cannot compile. The compiler reports `error: cannot order dictionary and dictionary`.

```typst
#let a = (("a": 2), ("a": 3)).sorted(item => item.at("a"))
```

However, with the key function applied here, sorting should be possible.

### Typst version

- [X] I am using the latest version of Typst

**Comment by iamkroot** [comment]
Was this ever actually fixed? I still see this error on v0.8.0

**Comment by laurmaedje** [comment]
The error message is very confusing, that's the real problem. The problem is that the argument is named and called `key`. 

```typst
#let a = (("a": 2), ("a": 3)).sorted(key: item => item.at("a"))
```

However, the check for extraneous arguments happens _after_ the sorting and that fails when no key is given. I'll repurpose this issue to track the diagnostics issue.
